services/navigation/builtin/builtin_test.go

```diff
@@ -43,157 +52,294 @@ func setupNavigationServiceFromConfig(t *testing.T, configFilename string) (navi
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, navMode, test.ShouldEqual, navigation.ModeManual)
 
+	// Mocked nav service location is a specific geo coordinate
 	geoPose, err := ns.Location(ctx, nil)
 	test.That(t, err, test.ShouldBeNil)
 	expectedGeoPose := spatialmath.NewGeoPose(geo.NewPoint(40.7, -73.98), 25.)
 	test.That(t, geoPose, test.ShouldResemble, expectedGeoPose)
 
+	// Waypoints default to an empty slice
 	wayPt, err := ns.Waypoints(ctx, nil)
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, wayPt, test.ShouldBeEmpty)
 
+	// Waypoints are able to be added
 	pt := geo.NewPoint(0, 0)
 	err = ns.AddWaypoint(ctx, pt, nil)
 	test.That(t, err, test.ShouldBeNil)
 
+	// Added Waypoints are returned from Waypoints
 	wayPt, err = ns.Waypoints(ctx, nil)
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, len(wayPt), test.ShouldEqual, 1)
 
+	// Added Waypoints can be removed
 	id := wayPt[0].ID
 	err = ns.RemoveWaypoint(ctx, id, nil)
 	test.That(t, err, test.ShouldBeNil)
+
+	// Removed Waypoints are no longer returned from Waypoints
 	wayPt, err = ns.Waypoints(ctx, nil)
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, len(wayPt), test.ShouldEqual, 0)
 
+	// Calling RemoveWaypoint on an already removed waypoint doesn't return an error
+	err = ns.RemoveWaypoint(ctx, id, nil)
+	test.That(t, err, test.ShouldBeNil)
+
+	// The mocked Nav service detects a single obstacle
 	obs, err := ns.GetObstacles(ctx, nil)
 	test.That(t, len(obs), test.ShouldEqual, 1)
 	test.That(t, err, test.ShouldBeNil)
 
+	// The motion config provides a single vision service
 	test.That(t, len(ns.(*builtIn).motionCfg.VisionServices), test.ShouldEqual, 1)
 }
 
 func TestStartWaypoint(t *testing.T) {
-	ctx := context.Background()
-	logger := golog.NewTestLogger(t)
+	t.Run("Calls motion.MoveOnGlobe for every waypoint", func(t *testing.T) {
+		callChan := make(chan moveOnGlobeReq)
+		injectMSensor := inject.NewMovementSensor("test_movement")
 
-	injectMS := inject.NewMotionService("test_motion")
-	cfg := resource.Config{
-		Name:  "test_base",
-		API:   base.API,
-		Frame: &referenceframe.LinkConfig{Geometry: &spatialmath.GeometryConfig{R: 100}},
-	}
+		injectMS := inject.NewMotionService("test_motion")
+		injectMS.MoveOnGlobeFunc = func(
+			ctx context.Context,
+			componentName resource.Name,
+			destination *geo.Point,
+			heading float64,
+			movementSensorName resource.Name,
+			obstacles []*spatialmath.GeoObstacle,
+			motionCfg *motion.MotionConfiguration,
+			extra map[string]interface{},
+		) (bool, error) {
+			callChan <- moveOnGlobeReq{
+				componentName:      componentName,
+				destination:        destination,
+				heading:            heading,
+				movementSensorName: movementSensorName,
+				obstacles:          obstacles,
+				motionCfg:          motionCfg,
+				extra:              extra,
+			}
+			return true, nil
+		}
+		ctx := context.Background()
+		logger := golog.NewTestLogger(t)
+		b := fakeBase(ctx, t, logger)
```

Comment:
I think these can all safely be moved outside the individual tests

Reply:
done